qt/qtsystrayicon.py

```diff
@@ -31,7 +31,10 @@
 import qttools
 qttools.registerBackintimePath('common')
 
+# Workaround until the codebase allows a single place to init all translations
 import tools
+tools.initiate_translation(None)
```

Comment:
The current code in `app.py` has the same structure (and is using the system language for all imports until `__main__` is executed), is this a bug then?

https://github.com/bit-team/backintime/blob/976081bbd3e38cee0e7f5efcffe0f7392777e76f/qt/app.py#L31-L37

and in `__main__`:

https://github.com/bit-team/backintime/blob/976081bbd3e38cee0e7f5efcffe0f7392777e76f/qt/app.py#L1945-L1948

Reply:
In other words my dilemma is:

- I cannot use the language from the config file directly after import because the config is not loaded (and cannot because of missing imports):
  `tools.initiate_translation(self.config.language())`

- I cannot move the init code from the `import tools...` to the `__init__` constructor because another import in the same file contains a function signature which already uses `_()` as default value and causes an "compilation" error:

https://github.com/bit-team/backintime/blob/976081bbd3e38cee0e7f5efcffe0f7392777e76f/qt/qtsystrayicon.py#L38
loads...
https://github.com/bit-team/backintime/blob/976081bbd3e38cee0e7f5efcffe0f7392777e76f/qt/logviewdialog.py#L28
which tries to validate
https://github.com/bit-team/backintime/blob/976081bbd3e38cee0e7f5efcffe0f7392777e76f/qt/messagebox.py#L55

- I cannot change the signature of `def info(text, title=_('Information'), widget_to_center_on=None):` because the default value needs to be there and translated

**So I suspect the structure of `app.py` is like my PR (causing the info message box to use the system language instead of the configured one) as best-possible workaround with the smallest impact (= wrong translations).**

I see no easy way to fix this :-(

Reply:
So if I get you right the best work-around is **to add** also this line into my `` constructor:

`tools.initiate_translation(self.config.language())`

I will send another commit ASAP...

PS: I think the fundamental problem is that the translations should be initialized **before** all imports happen to support correct translation of string default values in functions. Since our translation is loaded from a config file which requires heavy imports before this is not easily possible. And I see no simple solution for that and would say, above added line is the best I can do for now...

PS2: Thanks for pointing out the problem, no I understand the translation boot-strapping :-)

Reply:
Sorry for the noise but I think now that nothing must be changed because the language of the config file is already used to enable translation above (line 48 - 49) by loading the config (remember: `Config.__init__()` also calls `language_used = tools.initiate_translation(self.language())`):

https://github.com/bit-team/backintime/blob/976081bbd3e38cee0e7f5efcffe0f7392777e76f/qt/qtsystrayicon.py#L46-L60

So adding 

tools.initiate_translation(self.config.language())

into the line 56 would do the same again (unnecessarily) IMHO.

**So nothing todo but merge as-is?**

Reply:
I might have confused myself... :rofl: 
I'll take a look into it in the next hours to be sure. :face_with_spiral_eyes: